src/views/GrampsjsViewMap.js

```diff
@@ -389,12 +392,25 @@ export class GrampsjsViewMap extends GrampsjsView {
     if ('data' in data) {
       this.error = false
       this._dataEvents = data.data.filter(event => event.place)
+      this._minYear = this._getMinYear()
     } else if ('error' in data) {
       this.error = true
       this._errorMessage = data.error
     }
   }
 
+  _getMinYear() {
+    const years = this._dataEvents
+      ?.filter(event => event.place)
+      ?.map(event => getGregorianYears(event.date)?.[0])
+      ?.filter(y => y !== undefined)
+    let minYear = Math.min(...years)
+    const lastYear = new Date().getFullYear() - 1
+    minYear = Math.min(minYear, lastYear)
+    minYear = Math.max(minYear, 1) // disallow negative
```

Comment:
By the way, the map data and filtering code support BCE dates. Quite unlikely, of course, but I mention this in case we do decide to allow it in the future.

The only gotcha is that the year is offset by one because there’s no year 0, per ISO 8601. Currently this component is using −1 as a magic number for an unset year; we’d need to use something else like `undefined`.

Reply:
Yes you're right. I think it's pretty ok for Gramps Web, but of course not ok for OHM in general. So would be more elegant to handle in full generality in the future here as well.